app/profile_manager.py

```diff
@@ -0,0 +1,49 @@
+from flask import Blueprint, render_template, request, redirect, url_for
+from flask_login import login_required, current_user
+
+from sqlalchemy.exc import IntegrityError
+
+from . import db
+
+from .models import available_genders
+
+profile_manager_bp = Blueprint('profile_manager_bp', __name__)
+
+#TODO
+#Input validation
+#Image uploading
+#Consider using wtf or something like that
+
+@profile_manager_bp.route('/profile_manager', methods=['GET', 'POST'])
+@login_required
+def profile_manager():
+    if request.method == 'POST':
+        name = request.form.get("name")
+        gender = request.form.get("gender")
+        description = request.form.get("description")
+
+        gender_preferences = []
+        for g in available_genders:
+            if request.form.get(g):
+                gender_preferences.append(g)
+
+        lower_difference = request.form.get("lower_difference")
+        upper_difference = request.form.get("upper_difference")
+
+        current_user.profile.name = name
+        current_user.profile.gender = gender
+        current_user.profile.description = description
+        current_user.matching_preferences.gender_preferences = gender_preferences
+        current_user.matching_preferences.lower_difference = lower_difference
+        current_user.matching_preferences.upper_difference = upper_difference
+
```

Comment:
Prawdopodobnie by można było zrobić form_data = request.form.to_dict() i  potem bezpośrenio np. current_user.profile.name = form_data.get("name"). Co to tym sądzisz?

Reply:
To jest szczegół stylistyczny imo, można w każdej chwili zmienić jak uważasz że tak lepiej. Jeśli ostatecznie będziemy uzywać wtf do formularzy to wyglądałoby to current_user.profile.name = form.name.data